ddtrace/internal/datadog/profiling/dd_wrapper/include/sample_manager.hpp

```diff
@@ -19,12 +20,15 @@ class SampleManager
     static inline unsigned int max_nframes{ g_default_max_nframes };
     static inline SampleType type_mask{ SampleType::All };
     static inline std::mutex init_mutex{};
+    static inline size_t sample_pool_capacity{ g_default_sample_pool_capacity };
+    static inline std::unique_ptr<SynchronizedSamplePool> sample_pool{ nullptr };
```

Comment:
do we want this to exist across several translation units ?
should we not have a single object that references the pool that we can reset upon forking ?

Reply:
maybe? the underlying lock free queue doesn't have an api that lets the caller to take reference of the item in the container. All we do is `push` and `pop` basically.